lock.go

```diff
@@ -63,8 +63,6 @@ func readLock(r io.Reader) (*lock, error) {
 			v = gps.NewBranch(ld.Branch).Is(r)
 		} else if r == "" {
 			return nil, fmt.Errorf("lock file has entry for %s, but specifies no version", ld.Name)
-		} else {
-			v = r
```

Comment:
This is minor, but I personally don't like this change, style-wise. In some cases v is assigned twice, first time unnecessarily.

What was the lint message on this line? It looks like something's wrong for me:

```
lock.go:66: if block ends with a return statement, so drop this else and outdent its block
```

Not all `else if` blocks end with a return statement here, so following the advice above blindly would result in incorrect code. I wonder if that's an improvement opportunity for lint.